projects/cdk/components/dialog-host/dialog-host.component.ts

```diff
@@ -13,10 +14,16 @@ import {merge, Observable} from 'rxjs';
     animations: [TUI_PARENT_ANIMATION],
 })
 export class TuiDialogHostComponent<T extends TuiAriaDialogContext> {
-    readonly dialogs$ = merge(...this.dialogs);
+    readonly dialogs$ = combineLatest(this.allTypesDialogs).pipe(
+        map(allTypesDialogs =>
+            new Array<T>()
+                .concat(...allTypesDialogs)
+                .sort((a, b) => a.createdAt - b.createdAt),
+        ),
+    );
 
     constructor(
         @Inject(TUI_DIALOGS)
-        private readonly dialogs: ReadonlyArray<Observable<ReadonlyArray<T>>>,
+        private readonly allTypesDialogs: Observable<readonly T[]>[],
```

Comment:
Let's rename it to `dialogsByType` if you want to emphasize these are arrays of different type dialogs.

Reply:
done